hack-remove-global-override.service

```diff
@@ -0,0 +1,21 @@
+[Unit]
+Description=Remove hack flatpak global override
+DefaultDependencies=no
+Conflicts=shutdown.target
+Wants=local-fs.target
+After=local-fs.target
+
+# Only run on updates
+Before=multi-user.target systemd-update-done.service
+ConditionNeedsUpdate=|/etc
+ConditionNeedsUpdate=|/var
+
+ConditionPathExists=/var/lib/flatpak/overrides/global
+ConditionPathExists=/var/lib/flatpak/app/com.endlessm.HackComponents
```

Comment:
Are we going to keep `com.endlessm.HackComponents` installed on Hack 1.0 systems forever?

Reply:
The `com.endlessm.HackComponents` is needed for old hack toy apps, so I think we'll keep that until we replace all the hack toy apps or when we drop.

Reply:
OK – I think we need `Before=eos-updater-flatpak-installer.service` on this unit, assuming that would be the mechanism used to remove that Flatpak.

Otherwise – suppose we drop com.endlessm.HackComponents in 3.8.0. Then it's racy whether systemd checks this job's conditions or runs `eos-updater-flatpak-installer.service` which would (most likely) be the thing that uninstalls that Flatpak.

Reply:
I've just added that service to the list of Before in the Unit.